### PR TITLE
Automatically delete acme txt record for CloudXNS

### PR DESCRIPTION
**Pull request by kookxiang**
:D

### CODE REVIEW

File: dnsapi/dns_cx.sh [hunk 1]
```diff
@@ -82,7 +82,7 @@ existing_records() {
     return 1
   fi
   count=0
-  seg=$(printf "%s\n" "$response" | _egrep_o "{[^\{]*host\":\"$_sub_domain\"[^\}]*\}")
+  seg=$(printf "%s\n" "$response" | _egrep_o "{[^{]*host\":\"$_sub_domain\"[^}]*}")
```
Comment (Neilpang) [inline dnsapi/dns_cx.sh]: Why do you make this change ?
Is there any problem you see ?

It was added by me to make it work on freebsd and solaris.

Comment (kookxiang) [inline dnsapi/dns_cx.sh]: Hum.... It doesn't work on my ArchLinux. It cannot match the response text.
I'll make a test under freebsd later.

Comment (kookxiang) [inline dnsapi/dns_cx.sh]: It seems work well on my FreeBSD 11 VM too:
image

I restore that file, and it failed:
image

Comment (Neilpang) [inline dnsapi/dns_cx.sh]: sorry,  it was a bug.  your I'm fixing now.

Comment (Neilpang) [inline dnsapi/dns_cx.sh]: see my fix here:  https://github.com/Neilpang/acme.sh/commit/5be3f22d06cd62397395b0de4e76f3f1d3c27405

It's not working on solaris now, I'm still fixing.

Comment (Neilpang) [inline dnsapi/dns_cx.sh]: Hi,
I just fixed cx api, and tested on all the platforms.

Please rebase to the latest code and push again.

Thanks.

File: dnsapi/dns_cx.sh [hunk 2]
```diff
@@ -58,7 +58,15 @@ dns_cx_add() {
...
 dns_cx_rm() {
   fulldomain=$1
-
+  REST_API="$CX_Api"
+  if _get_root "$fulldomain"; then
+    record_id=""
+    existing_records "$_domain" "$_sub_domain"
+    if ! [ "$record_id" = "" ]; then
+      _rest DELETE "record/$record_id/$_domain_id" "{}"
+      _log "Deleted record ${fulldomain}"
```
Comment (Neilpang) [inline dnsapi/dns_cx.sh]: Use "_info"  instead.  do not use "_log"

**Pull request merged**